biblib/views/operations_view.py

```diff
@@ -204,34 +198,65 @@ def post(self, library):
                                      .format(request.data, error))
             return err(WRONG_TYPE_ERROR)
 
-        if data['action'] in ['union', 'intersection', 'difference']:
-            if 'libraries' not in data:
-                return err(NO_LIBRARY_SPECIFIED_ERROR)
-            if 'name' not in data:
-                data['name'] = 'Untitled {0}.'.format(get_date().isoformat())
-            if 'public' not in data:
-                data['public'] = False
+        action = data["action"]
 
-        if data['action'] == 'copy':
-            if 'libraries' not in data:
+        if action in ["union", "intersection", "difference"]:
+            if "libraries" not in data:
                 return err(NO_LIBRARY_SPECIFIED_ERROR)
-            if len(data['libraries']) > 1:
+            if "name" not in data:
+                data["name"] = "Untitled {0}.".format(get_date().isoformat())
+            if "public" not in data:
+                data["public"] = False
+        elif action == "copy":
+            if "libraries" not in data:
+                return err(NO_LIBRARY_SPECIFIED_ERROR)
+            if len(data["libraries"]) > 1:
                 return err(TOO_MANY_LIBRARIES_SPECIFIED_ERROR)
 
         lib_names = []
+
         with current_app.session_scope() as session:
             primary = session.query(Library).filter_by(id=library_uuid).one()
             lib_names.append(primary.name)
-            if 'libraries' in data:
-                for lib in data['libraries']:
-                    try:
-                        secondary_uuid = self.helper_slug_to_uuid(lib)
-                    except TypeError:
-                        return err(BAD_LIBRARY_ID_ERROR)
-                    secondary = session.query(Library).filter_by(id=secondary_uuid).one()
-                    lib_names.append(secondary.name)
-
-        if data['action'] == 'union':
+
+            if action == "empty":
```

Comment:
I really like this change to use `permission_check_<>`

Reply:
Thanks. It was pretty readable in the end.